src/server/db_slice.cc

```diff
@@ -1517,6 +1520,8 @@ void DbSlice::CallChangeCallbacks(DbIndex id, std::string_view key, const Change
   FetchedItemsRestorer fetched_restorer(&fetched_items_);
   std::unique_lock<LocalBlockingCounter> lk(block_counter_);
```

Comment:
Do we still need the `block_counter_` if we lock using a real lock?
Keep in mind that unregistering is very uncommon, we should not optimize for that use case

Reply:
We are not optimizing it, we get UB if we don't have it so yes we still need it. Also keep in mind that this doesn't lock anything, it merely increments an internal counter

Reply:
nit: unrelated to this PR - i think a betterlock to choose here is guard_lock or even scoped_ptr because this is what it is - a RAII guard.